### ISSUE TITLE
Pass actual dates in API calls in Search Console widgets

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#2237]
The Search Console `dashboard` and `pageDashboard` widgets should be migrated to pass `startDate` and `endDate` instead of `dateRange` with their API requests.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The following widgets should pass `startDate` and `endDate` (using the `core/user` `getDateRangeDates` selector) instead of `dateRange` to their `getReport` calls:
    * `DashboardPopularKeywordsWidget`
* The following widgets should pass `startDate` and `endDate` (using the `core/user` `getDateRangeDates` selector) instead of `dateRange` and `compareDateRanges` to their `getReport` calls:
    * `DashboardClicksWidget`
    * `DashboardImpressionsWidget`
* The dates sent should match (compare the API responses) what the server would have previously computed from the passed `dateRange` (and related) parameter (e.g. depending on the `offsetDays` value passed to `getDateRangeDates`).

## Implementation Brief

* Update the `DashboardPopularKeywordsWidget` component to use the `getDateRangeDates` selector from the user datastore to pull `startDate` and `endDate` dates. These dates must be set as `startDate` and `endDate` properties in the report args object instead of just `dateRange`.
* Update `DashboardClicksWidget` and `DashboardImpressionsWidget` components to use the `getDateRangeDates` selector from the user datastore to pull `compareStartDate` and `compareEndDate` dates using `compare: true` option. These dates must be set as `startDate` and `endDate` properties in the report args instead of just `dateRange`, plus `compareDateRanges: true` argument should be removed.

### Test Coverage

* Not needed.

### Visual Regression Changes

* Not expected.

## QA Brief

* Make sure the aforementioned widgets use the new date range dates instead of ranges.

## Changelog entry

* Move date awareness for Search Console API requests to the client, allowing for more contextual date information to be available.

**Comment by felixarntz** [comment]
@eugene-manuilov IB mostly ✅ , however I think it'd be worth adding that when calling `getDateRangeDates` with `compare: true`, it will also return `compareStartDate` and `compareEndDate` (which are the compare period, before the main period). To make that work, we'll need to pass `compareStartDate` as `startDate` to the API, so that the results basically encompass both ranges in one.

**Comment by eugene-manuilov** [comment]
@felixarntz yep, updated.

**Comment by felixarntz** [comment]
@benbowler There is some additional complexity here around making this work in Storybook, which we didn't consider originally. See #2459 for the related fixes to the AdSense date parsing logic. We should consider this here too and make sure the Storybook stories for those widgets still work, which will depend on the approach of #2459 and furthermore require a bit of extra work. Because of timeline and because this is not a user-facing enhancement, I'll remove it from the release to give us a bit more time.

Let's figure out a plan for #2459 and then we can act accordingly here (the same applies to #2236).

**Comment by benbowler** [comment]
Hey @felixarntz, I've had some time to look at this today. I've read through [2465](https://github.com/google/site-kit-wp/pull/2465/files) and can see your addition of the `referenceDate` attribute. With the Search Console I can't immediately see issues with the stories for them. If I add `referenceDate` based on the data it doesn't affect that data that is shown in the widgets.

What are the specific steps to get these widgets up to date with #2459 or can we close this ticket now?

**Comment by felixarntz** [comment]
@benbowler When you check the "Loaded" versions of the Storybook stories for the three widgets you've adjusted here, you'll see they all display the blue "Search Console Gathering Data" CTA which is displayed when there is no data in the response. This is due to the fixtures for these stories being now outdated, as the request arguments have changed (`startDate` and `endDate` instead of `dateRange`).

You'll need to update the `options` for these widget stories to match something in the format of how the widgets request their data, otherwise the `options` will basically be ignored by the component, causing that "empty" widget state. You can see how I made similar changes in #2459 to fix the same problem for the AdSense widget stories.

Basically, all the "Loaded" versions of the widgets in Storybook should display the regular UI with data available, how it is in `develop`.

**Comment by felixarntz** [comment]
@benbowler I just saw you moved this to Sprint 39. Let's not update this unless it was discussed/confirmed. We can remove it from the release as mentioned earlier, but this isn't blocked by anything, so I don't think it needs to wait. While #2236 is related, they don't need to be worked on at the same time, similarly to how #2235 was also implemented before, unrelated in timing.

**Comment by benbowler** [comment]
Hey @felixarntz, thanks for these extra details I'll take a look today and get an updated version into CR.

**Comment by benbowler** [comment]
Updated in the PR.

**Comment by eugene-manuilov** [comment]
QA :heavy_check_mark: Search Console widgets use date range dates now instead of range slugs.
